src/core/job.c

```diff
@@ -477,6 +478,15 @@ int job_type_merge_and_collapse(JobType *a, JobType b, Unit *u) {
         return 0;
 }
 
+static bool service_will_auto_restart(Unit *u) {
+        Service *s = SERVICE(u);
+
+        if (!s)
+                return false;
+
+        return (s->will_auto_restart || s->state == SERVICE_AUTO_RESTART);
+}
```

Comment:
For this I think you can reuse https://github.com/systemd/systemd/blob/97cff6cd5f58dd7262a0a7e0dec5f515b4fac93f/src/core/service.c#L1831-L1842

Reply:
I didn't reuse this because of the `assert`. In my case I don't want to fail if the unit is not a service, I want to return `false`.

Reply:
You can check that beforehand. It still feel better not to introduce functions that are named similar and behaves only slightly different.

Reply:
Actually I might be able to use 

https://github.com/systemd/systemd/blob/32e7970304805382526d72f5e51a970e5cce6ba4/src/core/unit.c#L3782-L3789

Reply:
Even better :) Thank you.